api/proto/teleport/legacy/types/types.proto

```diff
@@ -4744,6 +4744,98 @@ enum RequireMFAType {
   HARDWARE_KEY_TOUCH = 3;
 }
 
+// Plugin describes a single instance of a Teleport Plugin
+message PluginV1 {
+  // kind is the plugin resource kind.
+  string kind = 1;
+  // sub_kind is an optional resource subkind.
+  string sub_kind = 2;
+  // version is the resource version.
+  string version = 3;
+
+  // metadata is the resource metadata.
+  Metadata metadata = 4 [(gogoproto.nullable) = false];
+
+  PluginSpecV1 spec = 5 [(gogoproto.nullable) = false];
+
+  PluginStatusV1 status = 6 [(gogoproto.nullable) = false];
+
+  // credentials are "live" credentials to the 3rd party API.
+  // These are considered secrets.
+  PluginCredentialsV1 credentials = 7;
+}
+
+message PluginSpecV1 {
+  option (gogoproto.equal) = true;
+
+  // settings contain provider-specific plugin options.
```

Comment:
```suggestion
message PluginSpecV1 {
  // settings contain provider-specific plugin options.
```

Reply:
`Equal()` implementation will be used in subsequent PRs. We intend to compare the entire `Spec` to check if any settings have changed and whether we need to re-configure the plugin.

Reply:
ignore that then 👍